src/test/groovy/edu/ie3/datamodel/models/input/system/type/chargingpoint/ChargingPointTypeTest.groovy

```diff
@@ -0,0 +1,49 @@
+/*
+ * © 2024. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+package edu.ie3.datamodel.models.input.system.type.chargingpoint
+
+import edu.ie3.datamodel.models.ElectricCurrentType
+import edu.ie3.datamodel.models.StandardUnits
+import spock.lang.Specification
+import tech.units.indriya.quantity.Quantities
+
+class ChargingPointTypeTest extends Specification {
+
+  def "A ChargingPointType copy method should work as expected"() {
+    given:
+    def cpt = ChargingPointTypeUtils.HouseholdSocket
+
+    when:
+    def alteredUnit = cpt.copy()
+        .setElectricCurrentType(ElectricCurrentType.DC)
+        .setsRated(Quantities.getQuantity(2.4, StandardUnits.S_RATED))
```

Comment:
This should be 2.3, or?

Reply:
The household socket has 2.3 kVA originally (as per `ChargingPointTypeUtils`), and to check whether the copy builder works, we change that value to 2.4 kVA and test whether that value has been applied to the new `ChargingPointType`.

https://github.com/ie3-institute/PowerSystemDataModel/blob/667680f35d97170f0c83563c0323bcaf72a07bae/src/main/java/edu/ie3/datamodel/models/input/system/type/chargingpoint/ChargingPointTypeUtils.java#L40-L45

Reply:
True. It didn't get the setsRated..